pkg/ansible/watches/watches.go

```diff
@@ -149,50 +149,76 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // addRolePlaybookPaths will add the full path based on the current dir
 func (w *Watch) addRolePlaybookPaths() {
-	w.Playbook = getFullPath(w.Playbook)
-	w.Role = getFullRolePath(w.Role)
+	if len(w.Playbook) > 0 {
+		w.Playbook = getFullPath(w.Playbook)
+	}
+
+	if len(w.Role) > 0 {
+		possibleRolePaths := getPossibleRolePaths(w.Role)
+		for _, possiblePath := range possibleRolePaths {
+			if _, err := os.Stat(possiblePath); err == nil {
+				w.Role = possiblePath
+				break
+			}
+		}
+	}
 	if w.Finalizer != nil && len(w.Finalizer.Role) > 0 {
-		w.Finalizer.Role = getFullRolePath(w.Finalizer.Role)
+		possibleRolePaths := getPossibleRolePaths(w.Finalizer.Role)
+		for _, possiblePath := range possibleRolePaths {
+			if _, err := os.Stat(possiblePath); err == nil {
+				w.Finalizer.Role = possiblePath
+				break
+			}
+		}
 	}
 	if w.Finalizer != nil && len(w.Finalizer.Playbook) > 0 {
 		w.Finalizer.Playbook = getFullPath(w.Finalizer.Playbook)
 	}
 }
 
-// getFullPath will return a valid full path for the playbook
+// getFullPath returns an absolute path for the playbook
 func getFullPath(path string) string {
 	if len(path) > 0 && !filepath.IsAbs(path) {
 		return filepath.Join(projutil.MustGetwd(), path)
 	}
 	return path
 }
 
-// getFullRolePath will return a valid full path for the role
-func getFullRolePath(path string) string {
-	if len(path) > 0 && !filepath.IsAbs(path) {
-		envVar, ok := os.LookupEnv("ANSIBLE_ROLES_PATH")
-		if ok && len(envVar) > 0 {
-			// Check all values informed (e.g. path1/roles:path2/roles:/absolute/path3/roles)
-			result := strings.Split(envVar, ":")
-			for i := range result {
-				// Check if the role can be found in the envVar + role path
-				infoPath := filepath.Join(result[i], path)
-				if _, err := os.Stat(infoPath); err == nil {
-					return infoPath
-				}
-
-				// Check if the role can be found in the envVar + roles + role path
-				infoPathWithRolesDir := filepath.Join(result[i], "roles", path)
-				if _, err := os.Stat(infoPathWithRolesDir); err == nil {
-					return infoPathWithRolesDir
-				}
+// getFullRolePath returns a full path to the role directory
```

Comment:
comment should match the method name

Reply:
and the description should match what the method does